protocol/client/database.go

```diff
@@ -195,11 +201,16 @@ func (db SQLLiteDatabase) SaveContacts(contacts []Contact) (err error) {
 				return err
 			}
 		}
-		id := fmt.Sprintf("%s:%d", contacts[i].Name, contacts[i].Type)
+		id := contactID(contacts[i])
 		_, err = stmt.Exec(id, contacts[i].Name, contacts[i].Type, contacts[i].State, contacts[i].Topic, pkey)
 		if err != nil {
 			return err
 		}
+		// to avoid unmarshalling null into sql.NullInt64
+		_, err = history.Exec(id, 0)
```

Comment:
Should be wrapped in a transaction with `stmt.Exec` above or we don't care and entry in the history table is optional?

Reply:
history is a statement created with tx.Prepare. should be a part of the same transaction